packages/core/src/sync/index.ts

```diff
@@ -1026,7 +1035,7 @@ export async function* localHistoricalSyncGenerator({
      * time spent syncing ≈ time before indexing function feedback.
      */
     const interval: Interval = [
-      fromBlock,
+      Math.min(fromBlock, hexToNumber(historicalLast.number)),
```

Comment:
This wasn't required to fix the bug, and may be redundant, but it is a nice check to make sure the interval is always ordered correctly